pkg/pool-weighted/test/LiquidityBootstrappingPool.test.ts

```diff
@@ -1,2 +1,161 @@
+    itBehavesAsWeightedPool(3);
+  });
+
+  let trader: SignerWithAddress, lp: SignerWithAddress, owner: SignerWithAddress;
+
+  before('setup signers', async () => {
+    [, lp, trader, owner] = await ethers.getSigners();
+  });
+
+  let tokens: TokenList;
+
+  sharedBeforeEach('deploy tokens', async () => {
+    tokens = await TokenList.create(['MKR', 'DAI', 'SNX', 'BAT'], { sorted: true });
+    await tokens.mint({ to: [lp, trader], amount: fp(200) });
+  });
+
+  let pool: WeightedPool;
+  const weights = [fp(0.3), fp(0.55), fp(0.1), fp(0.05)];
+  const initialBalances = [fp(0.9), fp(1.8), fp(2.7), fp(3.6)];
+
+  sharedBeforeEach('deploy pool', async () => {
+    const params = { tokens, weights, owner, lbp: true, swapEnabledOnStart: true };
+    pool = await WeightedPool.create(params);
+  });
+
+  /*const initializePool = () => {
+    sharedBeforeEach('initialize pool', async () => {
+      await pool.init({ initialBalances, recipient: lp });
+    });
+  };*/
+
+  describe('weights', () => {
+    it('sets token weights', async () => {
+      const normalizedWeights = await pool.getNormalizedWeights();
+
+      // Need to decrease precision
+      expect(normalizedWeights).to.equalWithError(pool.normalizedWeights, 0.0001);
+    });
+  });
+
+  describe('public swap', () => {
+    sharedBeforeEach('deploy tokens', async () => {
+      const action = await actionId(pool.instance, 'setPublicSwap');
+      await pool.vault.grantRole(action, owner);
+    });
+
+    it('Owner can pause swaps', async () => {
+      await pool.setPublicSwap(owner, false);
+
+      expect(await pool.instance.swapEnabled()).to.be.false;
+
+      await expect(pool.swapGivenIn({ in: 1, out: 0, amount: fp(0.1) })).to.be.revertedWith('SWAPS_PAUSED');
+
+      await pool.setPublicSwap(owner, true);
+
+      expect(await pool.instance.swapEnabled()).to.be.true;
+    });
+
+    it('LP cannot pause swaps', async () => {
+      await expect(pool.setPublicSwap(lp, false)).to.be.revertedWith('SENDER_NOT_ALLOWED');
+    });
+  });
+
+  describe('joins', () => {
+    sharedBeforeEach('initialize pool', async () => {
+      await pool.init({ recipient: owner, initialBalances });
+
+      //const action = await actionId(pool.instance, '_onJoinPool');
+      //await pool.vault.grantRole(action, owner);
+    });
+
+    it('no public LPs', async () => {
+      await expect(pool.joinGivenIn({ recipient: lp, amountsIn: initialBalances })).to.be.revertedWith(
+        'SENDER_NOT_ALLOWED'
+      );
+    });
+
+    it('allows owner to join', async () => {
+      await expect(pool.joinGivenIn({ recipient: owner, from: owner, amountsIn: initialBalances })).to.not.be.reverted;
+    });
+  });
+
+  describe('update weights', () => {
+    sharedBeforeEach('deploy tokens', async () => {
+      const action = await actionId(pool.instance, 'updateWeightsGradually');
+      await pool.vault.grantRole(action, owner);
+    });
+
+    it('only allows owner to update weights', async () => {
```

Comment:
```suggestion
    it('non-owner cannot update weights', async () => {
```

I find it's better to say what the test actually does ('if `other` tries to update it reverts'), than what it attempts to prove ('only the owner can update'). It makes the tests easier to read, and the test case summary more honest.